src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexingService.cs

```diff
@@ -53,193 +61,173 @@ public class ElasticIndexingService
             _indexingTaskManager = indexingTaskManager;
             _elasticConnectionOptions = elasticConnectionOptions.Value;
             _siteService = siteService;
+            _contentManager = contentManager;
+            _contentItemIndexHandlers = contentItemIndexHandlers;
+            _store = store;
             _contentDefinitionManager = contentDefinitionManager;
             _logger = logger;
         }
 
-        public async Task ProcessContentItemsAsync(string indexName = default)
+        public async Task ProcessContentItemsAsync(params string[] indexNames)
         {
             if (!_elasticConnectionOptions.FileConfigurationExists())
             {
                 return;
             }
 
+            var indexSettingsList = (await _elasticIndexSettingsService.GetSettingsAsync()).ToList();
```

Comment:
You don't need ToList() here unless you want to use Insert, Add, Remove to a list. Then you can use the IEnumerable.

Reply:
var indexSettingsList = (await _elasticIndexSettingsService.GetSettingsAsync())

Reply:
Yes. But here I think it's better to convert to post because we'll need to do multiple check like checking count and then iterate over the correction. With to list, we'll avoid having to enumerate using Any(). Not huge advantage.

Reply:
I can see that you added it intentionally. This will allocate into a List from the IEnumerable. I'm not sure if it makes sense just to be able to use .Count afterward.

Reply:
I'll do a performance test tomorrow so I can see which is the best approach. It's worth understanding and explaining.

Reply:
It will surely copy the IEnumerable to the List
https://stackoverflow.com/questions/15516462/is-there-a-performance-impact-when-calling-tolist

Reply:
From memory we move away from Lists and even started using Dictionaries where we could here.

Reply:
I think I agree with you. But I am going to test it out anyway. 

I think if we use .Count() on IEnumerable, then it's probably best to use ToList() as .Count() will have to enumerate all the collection + will enumerate again later on. But I think when using .Any(), we'll enumerate the first record and then enumerate all the collection later without having to allocate. 

In these theory cases, I like to do performance test just to understand the difference and also understand the impact just for info. I'll post here the results tomorrow just FYI

Reply:
Also, a thing to keep in mind is that iterating over a large IEnumerable is significantly slower that List. Paying the allocation price to gain performance sometimes makes sense too. Anyway, I'll do some benchmarks tomorrow

Reply:
Ok, let's see the performance test.

Reply:
From what I've understood when I've read Mark Gravel stackoverflow 
https://stackoverflow.com/questions/305092/which-method-performs-better-any-vs-count-0
Use .Count when you already have an ICollection<T> but of course use .Any() if you don't. It's just that it's useless to do .Any() when the Collection already has .Count property on it. It's there to be used. Not to confound with .Count() which is not better than .Any().

Reply:
In general with IEnumerable<T> : stick with Any()
I would be surprised that it meant that we should move to .ToList() everywhere then.

Reply:
Yes 100%. Only use Count and List when available. Any() otherwise

Reply:
Ok so enough for today. I will wait for the perf test and continue reviewing when I can.